utils/webservice.go

```diff
@@ -214,14 +214,19 @@ func (hcs *HTTPConcertoservice) GetFile(path string, directoryPath string) (stri
 	defer response.Body.Close()
 	log.Debugf("Status code:%d message:%s", response.StatusCode, response.Status)
 
-	r, err := regexp.Compile("filename=\\\"([^\\\"]*){1}\\\"")
-	if err != nil {
-		return "", response.StatusCode, err
-	}
+	realFileName := ""
+	if directoryPath != "" && fileName == "" {
+		r, err := regexp.Compile("filename=\\\"([^\\\"]*){1}\\\"")
+		if err != nil {
+			return "", response.StatusCode, err
+		}
 
-	// TODO check errors
-	fileName := r.FindStringSubmatch(response.Header.Get("Content-Disposition"))[1]
-	realFileName := fmt.Sprintf("%s/%s", directoryPath, fileName)
+		// TODO check errors
+		fileName = r.FindStringSubmatch(response.Header.Get("Content-Disposition"))[1]
+		realFileName = fmt.Sprintf("%s/%s", directoryPath, fileName)
+	} else {
+		realFileName = fileName
+	}
```

Comment:
I would simplify and use directoryPath as filePath (dir path  + filename) when the url does not include it and directoryPath is not an existing directory. That way you do not need to change the interface for the method